### PR TITLE
feat(cli): G4 — cortex release, thin 4-surface deploy orchestrator + version-skew guard

### LINKED ISSUE #1120: G4: cortex release — orchestrate the 4 deploy surfaces
Refs #1116. A cortex release spans 4 independent deploy surfaces (bot via `arc upgrade`; dashboard, MC API worker, registry via 3× `wrangler`), none orchestrated — a bot-only upgrade leaves a stale dashboard/API/registry. Build a `cortex release` orchestrator (or checklist command) that sequences them with a version-skew guard. See audit §hotspot 5.

### PR DESCRIPTION
**Pull request by mellanon**
## Summary

- Adds `cortex release` — a thin checklist + guard command for the 4 independent cortex deploy surfaces (bot/dashboard/api/registry)
- Dry-run by default: prints the ordered plan with exact commands, no shell execution; `--apply` runs non-prod surfaces; `--apply --include-prod` also runs the two prod `wrangler --env production` surfaces with an explicit DEPLOYING TO PRODUCTION notice
- Version-skew guard: probes `/api/health` on `cortex.meta-factory.ai` and `network.meta-factory.ai`; reports ok/skew/unknown per surface without blocking the plan
- 33 hermetic tests with injected shell runner + HTTP fetcher — zero real deploys or network calls

## Design (separation-of-concerns adherence)

- **No daemon / no process orchestration** — `cortex release` is a single CLI invocation that prints a plan and optionally shells out to `arc`/`wrangler`/`bun build`. Exits when done.
- **Dry-run by default** — bare `cortex release` never runs a command; `--apply` is explicit opt-in.
- **Prod gate** — `api` and `registry` surfaces (`wrangler --env production`) require BOTH `--apply` AND `--include-prod`; otherwise printed as "skipped" in the plan.
- **Version source** — reads `arc-manifest.yaml` `version:` field verbatim (no new infra).
- **Commands verbatim** — `arc upgrade Cortex`, `bun build src/surface/mc/dashboard-v2/…`, `bunx wrangler pages deploy … --project-name grove-dashboard`, `wrangler deploy --env production --config src/surface/mc/worker/wrangler.toml`, `wrangler deploy --env production --config src/services/network-registry/wrangler.toml`.

## Files

- `src/cli/cortex/commands/release.ts` — implementation (330 LOC)
- `src/cli/cortex/commands/__tests__/release.test.ts` — 33 tests
- `src/cortex.ts` — wired using the existing passthrough commander pattern (same as `stack`/`network`/`offer`/`creds`)

## Test plan

- [x] `bun test src/cli/cortex/commands/__tests__/release.test.ts` → 33 pass, 0 fail
- [x] `bun test src/` → 7260 pass, 0 fail (full suite, no regressions)
- [x] `bunx tsc --noEmit` → clean
- [x] carve-out gate (`check-carveouts.sh`) → PASS
- [x] `bunx eslint --quiet` on touched files → clean

Closes #1120
Refs #1116

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cli/cortex/commands/__tests__/release.test.ts [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+    // Must have called wrangler deploy --env production for registry
+    const registryCalls = shell.calls.filter(
+      (c) => c.argv.includes("wrangler") &&
+              c.argv.includes("deploy") &&
+              c.argv.includes("--env") &&
+              c.argv.includes("production") &&
+              !c.argv.includes("pages"),
+    );
+    // At least 2 wrangler prod calls (api + registry) — could be same argv shape
+    expect(registryCalls.length).toBeGreaterThanOrEqual(1);
```
Comment (mellanon) [inline src/cli/cortex/commands/__tests__/release.test.ts:340]: **[nit/quality]** Comment says "at least 2 wrangler prod calls (api + registry)" but the assertion is `>= 1` — weaker than the comment promises.

The comment sets a stricter expectation than the assertion enforces. If only one prod call happened (e.g. if registry was silently skipped), the test would still pass. Consider tightening to `expect(registryCalls.length).toBeGreaterThanOrEqual(2)`, or add a separate assertion that explicitly checks for a registry-specific call. As-is, this assertion cannot distinguish "api + registry both deployed" from "only api deployed".

File: src/cli/cortex/commands/__tests__/release.test.ts [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+
+    expect(result.exitCode).toBe(0);
+    expect(shell.calls.length).toBe(1);
+    const call = shell.calls[0]!;
+    expect(call.argv).toContain("wrangler");
+    expect(call.argv).toContain("deploy");
+    expect(call.argv).toContain("--env");
+    expect(call.argv).toContain("production");
+  });
+});
```
Comment (mellanon) [inline src/cli/cortex/commands/__tests__/release.test.ts:609]: **[nit/quality]** Missing test: `--surfaces api --apply` WITHOUT `--include-prod` should result in zero shell calls (api is a prod surface, skipped without the flag).

The suite covers `--surfaces api --apply --include-prod` (line 575) but not the symmetric negative case. A test like:

```ts
test("--surfaces api --apply without --include-prod: zero shell calls (prod surface skipped)", async () => {
  const shell = mockShell(() => ({ exitCode: 0, stdout: "", stderr: "" }));
  __setShellRunnerForTests(shell.runner);
  __setFetcherForTests(mockFetcher(() => healthDown()).fetcher);
  const result = await dispatchRelease(["--apply", "--surfaces", "api"]);
  expect(result.exitCode).toBe(0);
  expect(shell.calls).toHaveLength(0); // prod gated, nothing runs
});
```

would lock in the gate behaviour for the targeted-surface path. The implementation is correct (`runSurface` returns `skipped` when `isProd && !includeProd`), but the test gap means a future regression could silently bypass the guard.

File: src/cli/cortex/commands/release.ts [hunk 3]
```diff
@@ -0,0 +1,645 @@
...
+  try {
+    return join(dirname(fileURLToPath(import.meta.url)), "../../../..");
+  } catch {
+    // During tests, import.meta.url may not be file://; walk up from __dirname
+    return join(__dirname, "../../../..");
+  }
+}
+
+/** Read the `version:` field from arc-manifest.yaml. */
+export function readIntendedVersion(): string {
```
Comment (mellanon) [inline src/cli/cortex/commands/release.ts:164]: **[nit/duplication]** Third independent `arc-manifest.yaml` reader in the repo.

`readIntendedVersion()` here is the third standalone implementation:
- `src/bus/dispatch-handler.ts` → `getCortexVersion()` (with caching via `_cachedVersion`)
- `src/cortex.ts` → `getVersion()` (regex-on-readFileSync, same pattern)
- `src/cli/cortex/commands/release.ts` → `readIntendedVersion()` (this PR)

All three parse `^version:\s+"?([^"\n]+)"?` from the manifest. Not a blocker for this PR (each lives in a different dependency domain), but worth tracking a follow-up to consolidate into a shared `readManifestVersion(root: string)` utility in `src/common/` before the count grows further.

**Review by mellanon — COMMENTED** [review]
Lenses applied: CodeQuality, Architecture, Duplication. 3 findings (0 blockers, 0 majors, 3 nits).

lint gate: ci — Lint job: SUCCESS. Typecheck: SUCCESS. Test: SUCCESS. All CI gates green.

architecture-docs: CONTEXT.md (loaded), docs/architecture.md (not-found), CONTEXT-MAP.md (not-found)

**Thin-orchestrator check (CLAUDE.md):** `dispatchRelease` is a plain async function. Calls shell via injected `ReleaseShellRunner`, returns `ExitResult`. No daemon, no PID file, no FileWatcher, no KeepAlive loop. Exits when done. Passes.

**Prod-deploy safety:** Dry-run is the default (bare invocation never shells out). `--apply` alone runs only `bot` and `dashboard` — `api` and `registry` (`wrangler --env production`) are gated behind BOTH `--apply` AND `--include-prod`. `--include-prod` without `--apply` throws `CliArgsError` exit-code 2. The `PROD_REQUIRED` map and `runSurface` guard are correct. Passes.

**Hermetic tests:** `__setShellRunnerForTests` / `__setFetcherForTests` seams inject mocks; `afterEach` restores null. No real arc/wrangler/bun calls or network fetches in the 33-test suite. Production defaults (`defaultShellRunner` / `defaultFetcher`) are only wired when no injection is active. Seam naming is consistent with the `offer.ts` `__setArcRunnerForTests` precedent. Passes.

**Version-skew guard:** Reads `arc-manifest.yaml` verbatim; probes `cortex.meta-factory.ai` and `network.meta-factory.ai`; reports `ok`/`skew`/`unknown` without guessing. `bot` and `dashboard` correctly return `unknown` (no queryable endpoint). Surface failure → non-zero exit. Passes.

**Secrets:** No tokens printed, no env-var exposure in output paths.

Findings:
- nit (line 340, test): `>= 1` assertion weaker than the "at least 2" comment promises — see inline.
- nit (line 609, test): missing symmetric negative test `--surfaces api --apply` → zero shell calls — see inline.
- nit (line 164, impl): third independent `arc-manifest.yaml` reader; follow-up to consolidate — see inline.

verdict: blockers=0 majors=0 nits=3 — recommend: merge (posted as comment-review — bot account opened the PR; --request-changes blocked by GitHub)
